ManzanMessageFormatter Exeption

Hi there,
I've encountered a new problem with Manzan.

I tried the audit option, but then a Java exception occurred.

```
java.lang.NullPointerException: null                                                                                             
        at com.github.theprez.manzan.ManzanMessageFormatter.format(ManzanMessageFormatter.java:39) ~[manzan.jar:?]               
        at com.github.theprez.manzan.routes.ManzanGenericCamelRoute.lambda$configure$1(ManzanGenericCamelRoute.java:50) ~[manzan.
jar:?]                                                                                                                           
```

data.ini
```
[audit1]
type=audit
destinations=test_out
auditType=PASSWORD
fallbackStartTime=1
format=Violation type: $VIOLATION_TYPE_DETAIL$, username: $AUDIT_USER_NAME$
interval=5000
```
dests.ini
```
[test_out]
type=http
url=http://SAMPLE_IP:3020
httpMethod=POST
format={"violation_type": "$VIOLATION_TYPE_DETAIL$", "username": "$AUDIT_USER_NAME$"}
```


Can anyone help me out?

Thanks

Hi,
I got the format out from the data.ini file, but it's not working again.
I have the full console output for you. Doing this with the file output and not the HTTP output works fine.

```
Apache Camel version 3.14.8                                                                                                      
target URI: http://SAMPLE_IP:3020?httpMethod=POST                                                                          
[                          main] AbstractCamelContext           INFO  Routes startup (total:2 started:2)                         
[                          main] AbstractCamelContext           INFO      Started route1 (direct://test_out)                     
[                          main] AbstractCamelContext           INFO      Started audit_table:audit1 (timer://foo)               
[                          main] AbstractCamelContext           INFO  Apache Camel 3.14.8 (camel-1) started in 617ms (build:74ms 
init:430ms start:113ms)                                                                                                          
MERGE INTO MANZAN.AUDJRNTS tgt USING (VALUES('PASSWORD')) src(AUDTYPE) ON tgt.AUDTYPE = src.AUDTYPE WHEN MATCHED THEN   UPDATE SE
T TIME = TIMESTAMP('2025-05-22 07:42:16.939136') WHEN NOT MATCHED THEN   INSERT (AUDTYPE, TIME) VALUES('PASSWORD', TIMESTAMP('202
5-05-22 07:42:16.939136'))                                                                                                       
[l-1) thread #9 - RecipientList] DefaultErrorHandler            ERROR Failed delivery for (MessageId: 73C58825D9DEE3B-00000000000
00005 on ExchangeId: 73C58825D9DEE3B-0000000000000005). Exhausted after delivery attempt: 1 caught: java.lang.NullPointerExceptio
n                                                                                                                                                                                                                                                           
                                                                                                                                 
Message History (complete message history is disabled)                                                                           
---------------------------------------------------------------------------------------------------------------------------------
------                                                                                                                           
RouteId              ProcessorId          Processor                                                                        Elapse
d (ms)                                                                                                                           
[audit_table:audit1] [audit_table:audit1] [from[timer://foo?period=5000&synchronous=true]                                ] [     
  813]                                                                                                                           
        ...                                                                                                                      
[route1            ] [process1          ] [Processor@0x8ee77b3e                                                          ] [     
    0]                                                                                                                           
                                                                                                                                 
Stacktrace                                                                                                                       
---------------------------------------------------------------------------------------------------------------------------------
------ 
java.lang.NullPointerException: null                                                                                             
        at com.github.theprez.manzan.ManzanMessageFormatter.format(ManzanMessageFormatter.java:39) ~[manzan.jar:?]               
        at com.github.theprez.manzan.routes.ManzanGenericCamelRoute.lambda$configure$1(ManzanGenericCamelRoute.java:50) ~[manzan.
jar:?]                                                                                                                           
        at org.apache.camel.support.processor.DelegateSyncProcessor.process(DelegateSyncProcessor.java:65) ~[manzan.jar:?]       
        at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:477) ~[manza
n.jar:?]                                                                                                                         
        at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:193) ~[manzan.jar:?]
        at org.apache.camel.impl.engine.DefaultReactiveExecutor.schedule(DefaultReactiveExecutor.java:59) ~[manzan.jar:?]        
        at org.apache.camel.processor.MulticastProcessor.lambda$schedule$1(MulticastProcessor.java:348) ~[manzan.jar:?]          
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]                                    
        at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]                                                   
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]                            
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]                            
        at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (07-03-2024)]                                                           
[mel-1) thread #1 - timer://foo] TimerConsumer                  WARN  Error processing exchange. Exchange[73C58825D9DEE3B-0000000  
000000000]. Caused by: [org.apache.camel.CamelExchangeException - Multicast processing failed for number 0. Exchange[73C58825D9DE
E3B-0000000000000002]. Caused by: [java.lang.NullPointerException - null]]                                                       
org.apache.camel.CamelExchangeException: Multicast processing failed for number 0. Exchange[73C58825D9DEE3B-0000000000000002]. Ca
used by: [java.lang.NullPointerException - null]                                                                                 
        at org.apache.camel.processor.MulticastProcessor$MulticastReactiveTask.lambda$null$0(MulticastProcessor.java:567) ~[manza
n.jar:?]                                                                                                                         
        at org.apache.camel.AsyncCallback.run(AsyncCallback.java:44) ~[manzan.jar:?]                                             
        at org.apache.camel.impl.engine.DefaultReactiveExecutor$Worker.schedule(DefaultReactiveExecutor.java:193) ~[manzan.jar:?]
        at org.apache.camel.impl.engine.DefaultReactiveExecutor.schedule(DefaultReactiveExecutor.java:59) ~[manzan.jar:?]        
        at org.apache.camel.processor.MulticastProcessor.lambda$schedule$1(MulticastProcessor.java:348) ~[manzan.jar:?]          
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:522) ~[?:1.8.0]                                    
        at java.util.concurrent.FutureTask.run(FutureTask.java:277) ~[?:1.8.0]                                                   
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160) ~[?:1.8.0]                            
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:1.8.0]                            
        at java.lang.Thread.run(Thread.java:825) ~[?:2.9 (07-03-2024)]                                                           
Caused by: java.lang.NullPointerException                                                                                        
        at com.github.theprez.manzan.ManzanMessageFormatter.format(ManzanMessageFormatter.java:39) ~[manzan.jar:?]            
        at com.github.theprez.manzan.routes.ManzanGenericCamelRoute.lambda$configure$1(ManzanGenericCamelRoute.java:50) ~[manzan.
jar:?]                                                                                                                           
        at org.apache.camel.support.processor.DelegateSyncProcessor.process(DelegateSyncProcessor.java:65) ~[manzan.jar:?]       
        at org.apache.camel.processor.errorhandler.RedeliveryErrorHandler$SimpleTask.run(RedeliveryErrorHandler.java:477) ~[manza
n.jar:?]                                                                                                                         
        ... 8 more                                                                                                              
                                                                                                            
```

--- a/camel/src/main/java/com/github/theprez/manzan/routes/event/AuditLog.java
+++ b/camel/src/main/java/com/github/theprez/manzan/routes/event/AuditLog.java
@@ -43,31 +43,37 @@ public class AuditLog extends ManzanRoute {
                 .to("jdbc:jt400?outputType=StreamList")
                 .process(exchange -> {
                     List<Map<String, Object>> rows = exchange.getIn().getBody(List.class);
-                    exchange.setProperty("resultSet", rows);
-
-                    // Find max ENTRY_TIMESTAMP
-                    Optional<Timestamp> maxTimestamp = rows.stream()
-                            .map(row -> (Timestamp) row.get("ENTRY_TIMESTAMP"))
-                            .filter(Objects::nonNull)
-                            .max(Comparator.naturalOrder());
-
-                    maxTimestamp.ifPresent(ts -> {
-                        String mergeQuery = String.format(
-                                "MERGE INTO MANZAN.AUDJRNTS tgt " +
-                                        "USING (VALUES('%s')) src(AUDTYPE) " +
-                                        "ON tgt.AUDTYPE = src.AUDTYPE " +
-                                        "WHEN MATCHED THEN " +
-                                        "  UPDATE SET TIME = TIMESTAMP('%s') " +
-                                        "WHEN NOT MATCHED THEN " +
-                                        "  INSERT (AUDTYPE, TIME) VALUES('%s', TIMESTAMP('%s'))",
-                                m_auditType, ts, m_auditType, ts
-                        );
-                        exchange.getIn().setBody(mergeQuery);
-                    });
-                    // If maxTimestamp is not present, set the body to null
-                    if (!maxTimestamp.isPresent()) {
+                    if (rows.isEmpty()){
                         exchange.getIn().setBody(null);
+                    } else {
+                        exchange.setProperty("resultSet", rows);
+
+                        // Find max ENTRY_TIMESTAMP
+                        Optional<Timestamp> maxTimestamp = rows.stream()
+                                .map(row -> (Timestamp) row.get("ENTRY_TIMESTAMP"))
+                                .filter(Objects::nonNull)
+                                .max(Comparator.naturalOrder());
+
+                        maxTimestamp.ifPresent(ts -> {
+                            String mergeQuery = String.format(
+                                    "MERGE INTO MANZAN.AUDJRNTS tgt " +
+                                            "USING (VALUES('%s')) src(AUDTYPE) " +
+                                            "ON tgt.AUDTYPE = src.AUDTYPE " +
+                                            "WHEN MATCHED THEN " +
+                                            "  UPDATE SET TIME = TIMESTAMP('%s') " +
+                                            "WHEN NOT MATCHED THEN " +
+                                            "  INSERT (AUDTYPE, TIME) VALUES('%s', TIMESTAMP('%s'))",
+                                    m_auditType, ts, m_auditType, ts
+                            );
+                            exchange.getIn().setBody(mergeQuery);
+                        });
+                        // If maxTimestamp is not present, set the body to null
+                        if (!maxTimestamp.isPresent()) {
+                            exchange.getIn().setBody(null);
+                        }
+
                     }
+
                 })
                 .choice()
                 .when(body().isNotNull())

